File: Atividade_8/Java/Controlador.java
import java.util.HashMap;
import java.util.List;

public class Controlador {

    private final HashMap<Departamento, List<Empregado>> funcionarios;

    public Controlador() {
        this.funcionarios = new HashMap<>();
    }

    public HashMap<Departamento, List<Empregado>> getFuncionarios() {
        return funcionarios;
    }

    public class Empregado {

        private final int id;
        private String nome;
        private String cargo;
        private double salario;
        private Departamento departamento;

        public Empregado(int id, String nome, String cargo, double salario, Departamento departamento) {
            this.id = id;
            this.nome = nome;
            this.cargo = cargo;
            this.departamento = departamento;
        }

        public Empregado(int id, String nome, String cargo, double salario) {
            this(id, nome, cargo, salario, null);
        }

        public int getId() {
            return id;
        }
        
        public String getNome() {
            return nome;
        }

        public void setNome(String nome) {
            this.nome = nome;
        }

        public double getSalario() {
            return salario;
        }

        public void setSalario(double salario) {
            this.salario = salario;
        }

        public String getCargo() {
            return cargo;
        }

        public void setCargo(String cargo) {
            this.cargo = cargo;
        }

        public Departamento getDepartamento() {
            return departamento;
        }

        public void setDepartamento(Departamento departamento) {
            this.departamento = departamento;
        }

    }

    public class Departamento {

        private int id;
        private String nome;
        private String localizacao;

        public void setId(int id) {
            this.id = id;
        }

        public void setLocalizacao(String localizacao) {
            this.localizacao = localizacao;
        }

        public void setNome(String nome) {
            this.nome = nome;
        }

        public int getId() {
            return id;
        }

        public String getNome() {
            return nome;
        }

        public String getLocalizacao() {
            return localizacao;
        }


        public void adicionarEmpregado(Empregado empregado) {}
        public void removerEmpregado(Empregado empregado) {}

    }

}

File: Atividade_8/C/controlador.c
typedef struct {
    int id;
    char[255] nome;
    char[255] localizacao;
    Empregado[255] empregados;
} Departamento;

void inicializarDepartamento(
    Departamento *departamento, 
    int id, 
    char[255] nome, 
    char[255] localizacao,
    Empregado[255] empregados;
) {}

int pegarIdDepartamento(Departamento *departamento) {}
char pegarNomeDepartamento(Departamento *departamento) {}
char pegarLocalizacaoDepartamento(Departamento *departamento) {}
Empregado *pegarEmpregadosDepartamento(Departamento *departamento) {}
void adicionarEmpregado(Departamento *departamento, Empregado *empregado) {}
void removerEmpregado(Departamento *departamento, Empregado *empregado) {}

typedef struct {
    int id;
    char[255] nome;
    char[255] cargo;
    float salario;
    Departamento *departamento;
} Empregado;

void inicializarDepartamento(
    Departamento *departamento, 
    int id, 
    char[255] nome, 
    char[255] localizacao,
    Empregado[255] empregados;
) {}

void inicializarEmpregado(
    Empregado *empregado,
    int id,
    char[255] nome,
    char[255] cargo,
    float salario,
    Departamento *departamento
) {}

int pegarIdEmpregado(Empregado *empregado) {}
char pegarNomeEmpregado(Empregado *empregado) {}
char pegarCargoEmpregado(Empregado *empregado) {}
float pegarSalarioEmpregado(Empregado *empregado) {}
Departamento *pegarDepartamentoEmpregado(Empregado *empregado) {}
void atualizarNomeEmpregado(Empregado *empregado, char[255] nome) {}
void atualizarCargoEmpregado(Empregado *empregado, char[255] cargo) {}
void atualizarSalarioEmpregado(Empregado *empregado, float salario) {}
void atualizarDepartamento(Empregado *empregado, Departamento *departamento) {}